products/batch_exports/backend/temporal/destinations/azure_blob_batch_export.py

```diff
@@ -0,0 +1,448 @@
+    bind_contextvars(
+        team_id=inputs.team_id,
+        destination="AzureBlob",
+        data_interval_start=inputs.data_interval_start,
+        data_interval_end=inputs.data_interval_end,
+    )
+    external_logger = EXTERNAL_LOGGER.bind()
+
+    azure_integration = await _get_azure_blob_integration(inputs.integration_id, inputs.team_id)
+
+    blob_key = get_blob_key(
+        prefix=inputs.prefix,
+        data_interval_start=inputs.data_interval_start,
+        data_interval_end=inputs.data_interval_end,
+        batch_export_model=inputs.batch_export_model,
+        file_format=inputs.file_format,
+        compression=inputs.compression,
+        is_splitting=inputs.max_file_size_mb is not None,
+    )
+
+    external_logger.info(
+        "Batch exporting range %s - %s to Azure Blob: %s",
+        inputs.data_interval_start or "START",
+        inputs.data_interval_end or "END",
+        blob_key,
+    )
+
+    async with Heartbeater():
+        queue = RecordBatchQueue(max_size_bytes=settings.BATCH_EXPORT_AZURE_BLOB_RECORD_BATCH_QUEUE_MAX_SIZE_BYTES)
+        producer = ProducerFromInternalStage()
+
+        assert inputs.batch_export_id is not None
+        producer_task = await producer.start(
+            queue=queue,
+            batch_export_id=inputs.batch_export_id,
+            data_interval_start=inputs.data_interval_start,
+            data_interval_end=inputs.data_interval_end,
+            max_record_batch_size_bytes=1024 * 1024 * 60,  # 60MB
+            stage_folder=inputs.stage_folder,
+        )
+
+        record_batch_schema = await wait_for_schema_or_producer(queue, producer_task)
+        if record_batch_schema is None:
+            external_logger.info(
+                "Batch export will finish early as there is no data matching specified filters in range %s - %s",
+                inputs.data_interval_start or "START",
+                inputs.data_interval_end or "END",
+            )
+            return BatchExportResult(records_completed=0, bytes_exported=0)
+
+        consumer = await AzureBlobConsumer.from_inputs(
+            inputs=inputs,
+            connection_string=azure_integration.connection_string,
+            max_concurrency=settings.BATCH_EXPORT_AZURE_BLOB_MAX_CONCURRENT_UPLOADS,
+        )
+
+        json_columns = ("properties", "person_properties", "set", "set_once")
+        if inputs.file_format.lower() == "jsonlines":
+            transformer = get_json_stream_transformer(
+                compression=inputs.compression,
+                include_inserted_at=True,
+                max_file_size_bytes=inputs.max_file_size_mb * 1024 * 1024 if inputs.max_file_size_mb else 0,
+            )
+        else:
+            transformer = ParquetStreamTransformer(
+                compression=inputs.compression,
+                include_inserted_at=True,
+                max_file_size_bytes=inputs.max_file_size_mb * 1024 * 1024 if inputs.max_file_size_mb else 0,
```

Comment:
No default file size limit since Azure blob limits ([190 TiB](https://learn.microsoft.com/en-us/azure/storage/blobs/scalability-targets)) far exceed realistic export sizes. Matches S3 [behavior](https://github.com/PostHog/posthog/blob/master/products/batch_exports/backend/temporal/destinations/s3_batch_export.py#L430).